### PR TITLE
Session object is not closed

### PR DESCRIPTION
**Pull request by eccles**
Problem:
The session object is not closed cleanly.

Solution:
Convert Archivist() as context manager.

Signed-off-by: Paul Hewlett <phewlett76@gmail.com>

### CODE REVIEW

File: archivist/archivistpublic.py [hunk 1]
```diff
@@ -109,6 +109,13 @@ def __getattr__(self, value: str) -> object:
         super().__setattr__(value, c)
         return c
 
+    def __enter__(self):
+        dummy = self.session
```
Comment (Serhiy1) [inline archivist/archivistpublic.py:113]: Why is this line needed?

File: functests/execsboms.py [hunk 2]
```diff
@@ -280,19 +280,22 @@ def test_sbom_upload_and_download(self):
...
-    def setUp(self):
-        super().setUp()
-        self.title = "TestSBOMWithApplication"
-        application = self.arch.applications.create(
-            DISPLAY_NAME,
-            CUSTOM_CLAIMS,
-        )
-        auth = (application["client_id"], application["credentials"][0]["secret"])
-        self.arch = Archivist(getenv("TEST_ARCHIVIST"), auth, verify=False)
+# class TestSBOMWithApplication(TestSBOM):
```
Comment (Serhiy1) [inline functests/execsboms.py:283]: Is this meant to be commented out?

File: archivist/archivistpublic.py (hunk 1 above)
Comment (eccles) [reply archivist/archivistpublic.py:113]: its not - good spot

File: functests/execsboms.py (hunk 2 above)
Comment (eccles) [reply functests/execsboms.py:283]: yes - its deadcode and I was testing if it was needed somewher and its not so deleted it

**Review by Serhiy1 — APPROVED** [review]
